### ISSUE TITLE
Model Weights

### ISSUE THREAD

**Issue by nishchay47b** [microsoft/table-transformer#8]
Hi,
Thank you for the paper and data, are there any plans to make model weights available.

Thanks

**Comment by bsmock** [comment]
Hi, thanks for the question. We do plan to release model weights but we do not currently have any timeline for this. At the moment the dataset is the only thing that is officially released. The training code is mostly stable, but we are still finalizing the paper and evaluation code. We want all of these to be stable and officially released before releasing any official model weights.

Best,
Brandon

**Comment by zjupx** [comment]
Any update for the Model Weights？

**Comment by phamquiluan** [comment]
For those who can't wait, I have some pre-trained weights here =)))
[https://github.com/phamquiluan/table-transformer](https://github.com/phamquiluan/table-transformer)

**Comment by Architectshwet** [comment]
When can we except an official documentation for doing inference to extract rows and columns from the table image using trained model?

**Comment by yixuanren** [comment]
> For those who can't wait, I have some pre-trained weights here =))) https://github.com/phamquiluan/table-transformer

Hi @phamquiluan , would you mind also releasing the final model.pth with more training-related metadata saved as well? Currently the pretrained model you released is only about the parameters of the model, while `model_state_dict`, `optimizer_state_dict` and `epoch` are missing (i.e. in the block https://github.com/phamquiluan/table-transformer/blob/fe04b4b92f8a537a8bee0437887619d930769b51/src/main.py#L294 the loading would fail with the ~110M model_%d.pth, and only the final ~320M model.pth has these information).

Thank you very much! Any suggestion or workaround would be appreciated.

**Comment by mazharsaif** [comment]
@bsmock any updates on expected release of official weights?

**Comment by bsmock** [comment]
Everyone, I'm happy to share that we released the detection model weights today. More updates are coming soon.

**Comment by phamquiluan** [comment]
Congrats for the CVPR @bsmock :smile:

**Comment by sanjaychamlagain123** [comment]
> For those who can't wait, I have some pre-trained weights here =))) https://github.com/phamquiluan/table-transformer

thanks!!  I tried inference in the images provided in your repo as well as other images..  It works perfectly for the samples provided by you but the bbox seems to squeeze in my samples..


<img width="491" alt="Screen Shot 2022-03-22 at 21 32 40" src="https://user-images.githubusercontent.com/52568735/159522799-326e62cd-2031-4b33-bd80-f182cba6c0a3.png">

**Comment by pengyuanzhang666** [comment]
I also encountered the same problem. Has it been solved now? @[sanjaychamlagain123](https://github.com/sanjaychamlagain123)

**Comment by ninikolov** [comment]
Hi, thanks for providing the weights for table detection. Is there any plan to release weights for the table recognition model? Would be great!

**Comment by sanjaychamlagain123** [comment]
@[pengyuanzhang666](https://github.com/pengyuanzhang666) i solved it using padding around the  images.

**Comment by bsmock** [comment]
Because we understand many people will try (or prefer) to use the model with tightly cropped table images, instead of the padded images in the data we released, we are currently discussing the best way to support this while maintaining reproducibility and properly documenting model performance given different padding/cropping strategies.

We plan to release official model weights as soon as this is resolved and document the different training approaches. Potentially we'll release multiple versions of the model trained with different cropping/padding strategies.

In the meantime, if you want to do inference on tightly cropped table images, two of your options are:
1. If you already have a trained model using the original version of the training code, you can simply pad your inference table images with whitespace or include more space around a table when you crop it.
2. You can modify the training code to do more cropping than the original version used and use this to fine-tune a previous model or re-train from scratch.

**Comment by mrvoh** [comment]
@bsmock Great work here! Any updates on when you will release the model weights for the TSR+FA model? It would be of great help.

Thanks already!

**Comment by bsmock** [comment]
Hey everyone, I'm happy to share that we were finally able to release the weights for the TSR + FA model today.

**Comment by bsmock** [comment]
Keep in mind that this version of the model works best with padding around the cropped tables, as the images in PubTables-1M are intentionally not cropped all the way to the table boundary in order to have more flexibility with augmentations at training time.

We plan to release additional models in the near future that are designed to work with tightly cropped table images.
